src/porepy/applications/verification_setups/manu_flow_incomp_frac.py

```diff
@@ -477,188 +657,192 @@ def set_md_grid(self) -> None:
+        source = internal_sources + external_sources
+        source.set_name("fluid sources")
+
+        return source
 
 
-class ModifiedSolutionStrategy(pp.fluid_mass_balance.SolutionStrategySinglePhaseFlow):
+class ManuIncompSolutionStrategy(pp.fluid_mass_balance.SolutionStrategySinglePhaseFlow):
+    """Modified solution strategy for the verification setup."""
 
     mdg: pp.MixedDimensionalGrid
+    """Mixed-dimensional grid."""
+
     darcy_flux: Callable[[list[pp.Grid]], pp.ad.Operator]
-    sol: StoreResults
+    """Method that returns the Darcy fluxes in the form of an Ad operator. Usually
+    provided by the mixin class :class:`porepy.models.constitutive_laws.DarcysLaw`.
+
+    """
+
+    exact_sol: ManuIncompExactSolution
+    """Exact solution object."""
+
+    fluid: pp.FluidConstants
+    """Object containing the fluid constants."""
+
+    plot_results: Callable
+    """Method to plot results of the verification setup. Usually provided by the
+    mixin class :class:`SetupUtilities`.
+
+    """
+
+    solid: pp.SolidConstants
+    """Object containing the solid constants."""
+
+    results: list[SaveData]
+    """List of SaveData objects."""
 
     def __init__(self, params: dict):
+        """Constructor for the class."""
 
-        # Parameters associated with the verification setup. The below parameters
-        # cannot be changed since they're associated with the exact solution.
-        # Normal permeability of 1/2 counteracts division by a/2 in the normal Darcy
-        # equation.
-        fluid = pp.FluidConstants({"compressibility": 0})
-        solid = pp.SolidConstants(
-            {"porosity": 0, "residual_aperture": 1, "normal_permeability": 1 / 2}
-        )
-        material_constants = {"fluid": fluid, "solid": solid}
-        required_params = {"material_constants": material_constants}
-        params.update(required_params)
         super().__init__(params)
 
-        self.solution: StoreResults
-        """Solution object that stores exact and approximated solutions and errors"""
-
-    def plot_results(self) -> None:
-        """Plotting results"""
+        self.exact_sol: ManuIncompExactSolution
+        """Exact solution object."""
 
-        self._plot_rock_pressure()
-        self._plot_fracture_pressure()
-        self._plot_interface_fluxes()
-        self._plot_fracture_fluxes()
+        self.results: list[SaveData] = []
+        """Results object that stores exact and approximated solutions and errors."""
 
-    def _plot_rock_pressure(self):
-        sd_rock = self.mdg.subdomains()[0]
-        p_num = self.sol.approx_rock_pressure
-        p_ex = self.sol.exact_rock_pressure
-        pp.plot_grid(
-            sd_rock, p_ex, plot_2d=True, linewidth=0, title="Rock pressure (Exact)"
-        )
-        pp.plot_grid(
-            sd_rock, p_num, plot_2d=True, linewidth=0, title="Rock pressure (MPFA)"
-        )
+        self.subdomain_darcy_flux_variable: str = "darcy_flux"
```

Comment:
Why is this needed as a variable?

Reply:
The idea here is to give keywords to secondary variables, e.g., subdomain Darcy fluxes. These are retrieved after convergence and stored in pp.STATE. Later, they are accessed by the data-saving classes.

Reply:
Fixed.